util/certs/certs.go

```diff
@@ -82,5 +82,24 @@ func DecodePrivateKeyPEM(encoded []byte) (*rsa.PrivateKey, error) {
 		return nil, nil
 	}
 
-	return x509.ParsePKCS1PrivateKey(block.Bytes)
+	key, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+
+	if err == nil {
+		return key, nil
+	}
+
+	// ParsePKCS1PrivateKey will fail with errors.New for many reasons
+	// including if the format is wrong, so we can retry with PKCS8
+	// https://golang.org/src/crypto/x509/pkcs1.go#L58
+	pkcs8Key, pkcs8Err := x509.ParsePKCS8PrivateKey(block.Bytes)
+	if pkcs8Err != nil {
+		// return the original error, since the pkcs8 error might be
+		// just a wrong format
+		return nil, err
+	}
+	key, ok := pkcs8Key.(*rsa.PrivateKey)
+	if !ok {
+		return nil, errors.New("Private key not in RAS format")
```

Comment:
I think this is a typo

```suggestion
		return nil, errors.New("Private key not in RSA format")
```

Reply:
But also it looks like from https://golang.org/src/crypto/tls/tls.go?#L254 this can be a `*rsa.PrivateKey`, `*ecdsa.PrivateKey` or a `ed25519.PrivateKey`, do we care?

Reply:
That's the reason for the type assertion. If it is something else than *rsa.PrivateKey, it returns an error.

Reply:
Yeah I meant why do we specifically want to exclude those two?

Reply:
that's true, there is no requirement to have a *rsa.PrivateKey because the x509 functions that are used support the three. I have removed this requirement.